ReactAndroid/src/main/java/com/facebook/react/views/text/ReactInlineBorderSpan.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package com.facebook.react.views.text;
+
+import android.graphics.Paint;
+import android.text.style.LineBackgroundSpan;
+import android.graphics.Canvas;
+import android.graphics.Paint;
+import android.graphics.RectF;
+import android.graphics.Color;
+
+/**
+ * We use a custom {@link LineBackgroundSpan}, because RN Text component can't render borders. Details here:
+ * https://github.com/Expensify/App/issues/4733
+ */
+public class ReactInlineBorderSpan implements LineBackgroundSpan, ReactSpan {
+  private final int effectiveStart;
+  private final int effectiveEnd;
+  private final int effectiveFontSize;
+
+  private final int backgroundColor = Color.parseColor("#f6f6f6");
+  private final int borderColor = Color.parseColor("#b3b8bc");
+  private final int borderRadius = 8;
+  private final int borderWidth = 2;
+
+  public ReactInlineBorderSpan(int effectiveFontSize, int effectiveStart, int effectiveEnd) {
+    this.effectiveFontSize = effectiveFontSize;
+    this.effectiveStart = effectiveStart;
+    this.effectiveEnd = effectiveEnd;
+  }
+
+  /**
+   * Calculates text start and end position per given lines text range
+   */
+  private int[] calculateBorderedTextCharRange(int lineStart, int lineEnd) {
+    /**
+     * If bordered text's position falls in lines size range return the position of bordered text's first char
+     * Otherwise return lines start position.
+     * 
+     * e.g. given paragraph below:
+     * This is a sample text [and the bordered | prependedTextStart -> 23, prependedTextEnd -> 39
+     * text starts and ends here].             | prependedTextStart -> 24, prependedTextEnd -> 65
+     */
+    int prependedTextStart = (lineStart < this.effectiveStart && lineEnd > this.effectiveStart) ? this.effectiveStart : lineStart;
+    int prependedTextEnd = (lineEnd < effectiveEnd) ? lineEnd : effectiveEnd;
+
+    return new int[]{prependedTextStart, prependedTextEnd};
```

Comment:
I'm a bit confused by the object syntax here. Wouldn't something similar to this work: `return new array<int>(prependedTextStart, prependedTextEnd);`

Reply:
Maybe I'm just unfamiliar with Java syntax, but I don't remember needing to use curly braces

Reply:
https://stackoverflow.com/questions/9114047/curly-braces-when-define-array